velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1213,6 +1213,25 @@ struct CurrentDateFunction {
   }
 };
 
+template <typename T>
+struct FromISO8601DateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Date>& result,
+      const arg_type<Varchar>& isoDateStr) {
+    int len = isoDateStr.size();
```

Comment:
int len = min(isoDateStr.size(), 10);
result = len < 10 ? DATE()->toDays(isoDateStr) : DATE()->toDays(isoDateStr.substr(0, 10))

I think you can work on the original StringViews without all the copies to std::string and substr on it.

Reply:
I'm a fan of ternary exps, though avoided to keep clarity/wasn't sure if favored. Condensed the code as such.

I initially tried working on the StringView but it didn't seem possible (doesn't seem to have methods like substr), so extracting the std::string seemed like the best bet.